src/components/views/projects/new/mdx_project_details_view.tsx

```diff
@@ -9,48 +9,85 @@ import { useDatabase } from '../../../../providers/database_provider'
 import StoreProvider from '../../../../providers/store_provider'
 import DOEProjectDetailsTemplate from '../../../../templates/doe_project_details.mdx'
 import { type Project } from '../../../../types/database.types'
+import { type Installer } from '../../../../types/installer.type'
 import { newProject, putProject } from '../../../../utilities/database_utils'
 import { hasErrors } from '../../../../utilities/validation_utils'
 
 type MdxProjectViewProps = Record<string, never>
 
 const MdxProjectView: React.FC<MdxProjectViewProps> = () => {
     const db = useDatabase()
-
     const navigate = useNavigate()
+    const [project, setProject] = useState<PouchDB.Core.PutDocument<Project>>()
+
+    useEffect(() => {
+        const fetchAndInitializeProject = async () => {
```

Comment:
Can this be refactored into a new function in `database_utils.ts`?

```ts
useEffect(() => {
  getLastModifiedInstaller(db).then(installer => {
    const project = newProject('')

    setProject({
      ...project,
      data_: {
        ...project.data_,
        installer,
      },
    })
  })
}, [db])
```

Reply:
Also, consider using the `find` function instead of the `allDocs` function. The `PouchDB.Find.FindRequest` type accepts both [`sort` and `limit` props](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/pouchdb-find/index.d.ts#L76-L94), so you can sort by `last_modified_at` in descending order and only fetch 1 PouchDB document.